### ISSUE TITLE
Dark mode forced by default and cant be updated

### ISSUE THREAD

**Issue by Ellosaur** [jgraph/drawio#3692]
### Preflight Checklist

* [x] I agree to follow the [Code of Conduct](https://github.com/jgraph/drawio/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
* [x] I have searched the issue tracker for a feature request that matches the one I want to file, without success.

**Describe the bug**
Pre-existing diagrams used in Confluence have all suddenly been set to Dark mode. Editing them, setting them back to Light mode and clicking Publish has no effect.

**To Reproduce**
1. Go to an existing draw.io diagram you have in any Confluence page
2. See it now presents in Dark mode
3. Click the pencil to edit it inline
4. Select Extras -> Appearance -> Light
5. Click Publish
6. See it still presents in Dark mode

**Expected behavior**
Diagrams should not be changed from their previous default (Light) and specifying Light mode explictly should work.

**Screenshots**
![issue](https://github.com/jgraph/drawio/assets/3505421/1fdaa94c-9a07-42ef-9043-28f8bd4f4bfb)

**draw.io version (In the Help->About menu of the draw.io editor):**

- draw.io version 21.5.1

**Desktop (please complete the following information):**
 - OS: Windows
 - Browser: Chrome
 - Version: 114.0.5735.134 (Official Build) (64-bit)

**Comment by davidjgraph** [comment]
Is this only in the viewer, or also in the editor?

**Comment by davidjgraph** [comment]
OK, we are looking as a priority. Please be patient while we investigate, all engineers will be on the task, so none to answer questions here.

**Comment by rrsela** [comment]
Editor can be set to use a non-system theme, while the viewer is fixed to the system theme.

**Comment by maciejadamski0** [comment]
I have that issue on my Mac with dark mode. Atlassian stack is setup to light version. I changed in top right corner from Dark to Light and saved the diagram and it's still back to dark. Also I'm trying Top Bar -> Extras -> Appearance -> Light and saved diagram and back to dark mode.

**Comment by davidjgraph** [comment]
We've reverted support for dark mode in jira and confluence cloud in our dev branch, that's undergoing dev testing now.

**Comment by davidjgraph** [comment]
The changes passed initial testing and review. We're now going to build and deploy to pre-production. There we'll run through the full regression testing.

**Comment by rrsela** [comment]
Looks good (had to clear the cache and delete all application data) - thanks!

**Comment by etricky** [comment]
CTRL+F5 on the pages and it is now fixed, thank you for the quick fix!

**Comment by ivoniin** [comment]
This was fast, thank you so much

**Comment by RichardFranke** [comment]
That was a very unpleasant surprise while I was doing a presentation, but glad it was fixed so fast. Thanks to all.

**Comment by maciejadamski0** [comment]
Thanks works. Command + f5 resolve problem in safari

**Comment by ChrisGoulding-Bango** [comment]
Thanks for the fast turnaround, fix confirmed here too.

**Comment by haferburg** [comment]
Fixed here as well, thanks!

**Comment by Ellosaur** [comment]
Great thank you for the fix

**Comment by patrickweegen** [comment]
Here it also works perfectly now

**Comment by davidjgraph** [comment]
Closing as all report as fixed.

One thing to note, in terms of the poor readability of the diagrams in dark mode. In dark mode we do adjust colours so that you can see them in both the viewer and the editor.

What happened yesterday was the CSS variable that atlassian provide flipped to give a black background, but the condition the viewer depended on didn't flip. This way you get black text on a black background on most diagrams.

We'll remove our dependency on their CSS so they can't do it to us again. Our dark mode does work, honest.. There will be update improvements to the colour inverting coming up that will improve on the current implementation. So, if you do want dark mode in jira or cloud, it will all just work (TM).

**Comment by itaiofirhailo** [comment]
hello, this issue is appearing for us now (Dec 29, 2024), exactly as originally described. can this issue be re-opened or should I open a new ticket?
We checked and in our case we have mixed results on multiple computers, both running chrome/confluence in light mode - one appears as normal, while the other reproduces this issue (all graphs in dark theme, including editor, changing to light + publish does not help).
